File: Python/wikiapp.py
from responder import WikiPageResponder, SearchResponder
from request_response import RequestContext


class WikiApp:
    def __init__(self, root_page):
        self.root_page = root_page

    def handle_request(self, request):
        if request.request_type == "GET":
            responder = WikiPageResponder()
        elif request.request_type == "POST":
            responder = SearchResponder()
        else:
            raise Exception('Method not supported')
        return responder.make_response(request, RequestContext(self.root_page))

File: Java/src/main/java/wikisearch/WikiApp.java
package wikisearch;

public class WikiApp {

    private WikiPage rootPage;

    public WikiApp(WikiPage rootPage) {
        this.rootPage = rootPage;
    }

    public Response handleRequest(Request request) {
        Responder responder;
        if (request.type.equals("GET")) {
            responder = new WikiPageResponder();
        } else if (request.type.equals("POST")) {
            responder = new SearchResponder();
        } else {
            throw new RuntimeException("Method not supported");
        }
        return responder.makeResponse(request, new RequestContext(this.rootPage));
    }
}
